numpyro/util.py

```diff
@@ -408,3 +408,121 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
         lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
     )
     return tree_map(lambda y: jnp.reshape(y, batch_shape + jnp.shape(y)[1:]), ys)
+
+
+def format_shapes(trace, *, title="Trace Shapes:", last_site=None, log_prob=False):
```

Comment:
It would be nice to expose this utility, at least to the docstring so that users know where to find this and know how to get the trace (especially for models with enumeration). What do you think is a good place for this? I'm thinking of renaming `Visualization` section and adding an entry for this utility there.

I think we can rename `log_prob` to [compute_log_prob](https://docs.pyro.ai/en/stable/_modules/pyro/poutine/trace_struct.html#Trace.compute_log_prob) which supports True/False/callable where callable can be used to filter the sites.

Reply:
Done, let me know if I did as you were expecting or if you wanted a whole new section.

Reply:
Thanks! LGTM.